wandb/apis/public.py

```diff
@@ -2531,6 +2531,46 @@ def aliases(self, aliases):
                 )
         self._aliases = aliases
 
+    @staticmethod
+    def expected_type(client, name, entity_name, project_name):
+        """Returns the expected type for a given artifact name and project"""
+        query = gql(
+            """
+        query Artifact(
+            $entityName: String!,
+            $projectName: String!,
+            $name: String!
+        ) {
+            project(name: $projectName, entityName: $entityName) {
+                artifact(name: $name) {
+                    artifactType {
+                        name
+                    }
+                }
+            }
+        }
+        """
+        )
+        if ":" not in name:
+            name += ":latest"
+
+        response = client.execute(
+            query,
+            variable_values={
+                "entityName": entity_name,
+                "projectName": project_name,
+                "name": name,
+            },
+        )
+
+        project = response.get("project")
+        if project is not None:
+            artifact = project.get("artifact")
+            if artifact is not None:
+                artifact_type = artifact.get("artifactType")
+                if artifact_type is not None:
+                    return artifact_type.get("name")
```

Comment:
Should we explicitly return `None` if any of the above are missing?

Reply:
Sure - I can make it explicit. Incoming.